server/application/rest/shelter.py

```diff
@@ -38,8 +74,14 @@ def shelter():
             status=HTTP_STATUS_CODES_MAPPING[ResponseTypes.SUCCESS]
         )
     elif request.method == "POST":
+        # Authorization check for adding shelters
+        if not is_authorized(repo, user_email, Resources.SHELTER):
```

Comment:
```suggestion
        if not is_authorized(repo, user_email, Resources.SYSTEM):
```
Only system admins are allowed to add shelters.